__tests__/fetcher/fetch-page-data.ts

```diff
@@ -23,3540 +37,491 @@ afterAll(() => {
   server.close()
 })
 
-test('User', async () => {
+function serverPostUuid(uuid: object) {
```

Comment:
Move this function to the end of the file. The tests should be first part of the file and the more low level helper function should be declared later.

Reply:
I would also rename the function to `givenApiReturnsUuid()` -> this makes the tests more readable. Or even better:

```
test("typename: User", async () => {
  givenApiReturns(uuidResponse({ __typename: "User", id: 1 }))
  ...
}

function givenApiReturns(json: any) {
  server.use(
    rest.post(endpoint, (_req, res, ctx) => ctx.json(json))
  )
}

function uuidResponse(uuid: object) {
  return { data: { uuid }}
}
```

Then you can use `givenApiReturns` also in tests for other api endpoints or when it returns an error.

Reply:
👍  renamed and moved, will seperate functions when the need arises so I know how that would make sense.